### DISCUSSION TITLE
Relative formatting for date only ("today", "tomorrow", "yesterday", ...)

### DISCUSSION THREAD

**Discussion by patrickhilker** [home-assistant/frontend#8191] [Dashboard]
Hey there!

I have several sensors created by the Google Calendar integration reporting a datetime. I would like to display the date as a relative date in the entities card, but without the time, as these dates represent the whole day.

Those are displayed for example as "in 5 hours" instead of "tomorrow". I would love to output the sensors dates as "today", "tomorrow", "yesterday", "in 5 days", "in 2 weeks" and so on.

Keep on the amazing work!

**Comment by spacegaier** [comment]
Some context for others reading this: If the `device_class` of those sensors is `timestamp` (and they probably should if they contain a timestamp) then the `entities` card allows you to define various render formats already: https://www.home-assistant.io/lovelace/entities/#format

But I think OP wants to have an additional one like `relative_date` (compared to the existing `relative`) that only considers the date portion for the relative time calculation. I like that and would have anyway raised a similar topic soon for also having `weekday` ("e.g "Monday") and `weekday_short` ("Mon") added. I might create a PR for those three.

**Reply by patrickhilker** [reply]
Absolutely - I also thought about a format called `relative_date`. Thanks in advance for your effort!

**Comment by ralphje** [comment]
This should also exist for entities with a device class of `date`, perhaps even as default (`relative` seems to be the default for `datetime` entities).

**Reply by jazzyisj** [reply]
I suggested this while back, it didn't get much traction.

https://community.home-assistant.io/t/date-device-class-relative-display/412469

**Comment by Sidelity** [comment]
I think this definitely should be added. Here's a real life example:

I have 2 template sensors:
1. Next alarm is using device class `Timestamp`. It formats nicely in the dashboard to "In X hours", or "Tomorrow", "In X days" etc.
2. Next watering day is using device class `Date`. It's not formatted nicely at all, and just displays the date, even if the date is tomorrow.

![](https://github.com/user-attachments/assets/15a39775-064c-4bc7-84ff-47645bae7b6a)
